src/views/AddItem.jsx

```diff
@@ -1,22 +1,36 @@
 import { useState } from 'react';
 import { addItem } from '../api/firebase';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data, setData }) {
 	const [daysUntilNextPurchase, setTimeFrame] = useState('7');
 	const [itemName, setItem] = useState('');
-	const [itemData, setData] = useState({});
 	const [error, setError] = useState(false);
+	const [duplicateError, setDuplicateError] = useState(false);
 	const [success, setSuccess] = useState(false);
 
+	// check for duplicate item in list
+	const isDuplicate = (data) =>
+		data.name.toLowerCase().replace(/ /g, '') ===
+		itemName.toLowerCase().replace(/ /g, '');
+
 	const handleSubmit = async (e) => {
 		e.preventDefault();
-		setData({ itemName, daysUntilNextPurchase });
+		const duplicateItem = data.some(isDuplicate);
+		console.log('data', data);
```

Comment:
the console.logging part could be removed to keep the code clean

Reply:
Good catch! I've removed them to clean it up.